SpatialGDK/Source/SpatialGDK/Public/Schema/RPCPayload.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+namespace SpatialGDK
+{
+
+struct RPCPayload
+{
+	RPCPayload() = delete;
+
+	RPCPayload(uint32 InOffset, uint32 InIndex, TArray<uint8> Data) : Offset(InOffset), Index(InIndex), PayloadData(Data)
+	{
+	}
+
+	RPCPayload(const Schema_Object* RPCObject)
+	{
+		Offset = Schema_GetUint32(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_OFFSET_ID);
+		Index = Schema_GetUint32(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_INDEX_ID);
+		PayloadData = SpatialGDK::GetBytesFromSchema(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_PAYLOAD_ID);
+	}
+
+	void WriteToSchemaObject(Schema_Object* RPCObject) const
+	{
+		Schema_AddUint32(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_OFFSET_ID, Offset);
+		Schema_AddUint32(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_INDEX_ID, Index);
+		AddBytesToSchema(RPCObject, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_PAYLOAD_ID, PayloadData.GetData(), sizeof(uint8) * PayloadData.Num());
+	}
+
+	uint32 GetOffset() const
+	{
+		return Offset;
+	}
+
+	uint32 GetIndex() const
+	{
+		return Index;
+	}
+
+	int64 CountDataBits() const
+	{
+		return PayloadData.Num() * 8;
+	}
+
+	const TArray<uint8>& GetData() const
+	{
+		return PayloadData;
+	}
+
+	TArray<uint8>& GetData()
+	{
+		return PayloadData;
+	}
+
+private:
+	uint32 Offset;
+	uint32 Index;
```

Comment:
Any reason to make these be accessed through accessor methods? We can probably just have them be public like our other types.